### PR TITLE
Updates for v3.24.0

### PR DESCRIPTION
**Pull request by mgleung**
## Description

Update the docs for releasing Calico v3.24.0

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: calico/_data/versions.yml [hunk 1]
```diff
@@ -1,36 +1,36 @@
-- title: v3.24.0-pre
+- title: v3.24.0
   note: ""
```
Comment (caseydavenport) [inline calico/_data/versions.yml:2]: Need to remove this line in order to get relnotes to show up

File: calico/_data/versions.yml [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
   manifests_url: https://raw.githubusercontent.com/projectcalico/calico/master
```
Comment (caseydavenport) [inline calico/_data/versions.yml:3]: need to change this manifests_url as well

should be /v3.24.0 instead of /master

**Review by caseydavenport — APPROVED** [review]
Couple comments but otherwise LGTM

File: calico/_data/versions.yml (hunk 1 above)
Comment (caseydavenport) [inline calico/_data/versions.yml:2]: Need to remove "note" still

**Comment by mgleung** [comment]
Closing in favor of https://github.com/projectcalico/calico/pull/6565

**Pull request closed without merging**